bdio-protobuf/src/main/proto/file_node_v1.proto

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2022 Synopsys Inc.
+ * http://www.synopsys.com/
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Synopsys ("Confidential Information"). You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Synopsys.
+ */
+syntax = "proto3";
+
+package bdio.v1;
+
+option java_multiple_files = true;
+option java_package = "com.blackducksoftware.bdio.proto.v1";
+option java_outer_classname = "FileNodeProto";
+
+/**
+ * See here for guidelines on avoiding breaking changes to message formats: https://developers.google.com/protocol-buffers/docs/proto#updating
+ * If those are not possible, or the message is sufficiently different, a new version should be created.
+ */
+message ProtoFileNode {
+
+  // The numeric ID of this file node.
+  int64 id = 1;
+
+  // The full URI of this file.
+  // Example: file:///Users/hjoe/dev/bdio/bdio-reactor/src/test/java/com/blackducksoftware/bdio2
+  string uri = 2;
+
+  // The path to this file relative to the root node of this scan.
+  // Example: bdio-reactor/src/test/java/com/blackducksoftware/bdio2/
+  string path = 3;
+
+  // The name of this file/directory/archive.
+  // Example: bdio2
+  string name = 4;
+
+  // The size of this file in bytes.
+  int64 size = 5;
+
+  // The file system type of this node.
+  // See com.blackducksoftware.scan.api.NodeType for possible values.
+  string file_system_type = 6;
+
+  // The ID of this node's parent.
+  optional int64 parent_id = 7;
+
+  // The path of archive that this file is contained in, if any.
+  // Example: gradle/wrapper/gradle-wrapper.jar!/
+  optional string archive_context = 8;
+
+  // The number of directories that are immediate children of this node.
+  optional int64 shallow_directory_count = 9;
```

Comment:
Similar comments as above.  Add documentation as to why these fields can be optional.

Reply:
documented.